python/semantic_kernel/connectors/ai/google/google_ai/services/google_ai_chat_completion.py

```diff
@@ -97,7 +116,40 @@ async def get_chat_message_contents(
         settings = self.get_prompt_execution_settings_from_settings(settings)
         assert isinstance(settings, GoogleAIChatPromptExecutionSettings)  # nosec
 
-        return await self._send_chat_request(chat_history, settings)
+        if (
+            settings.function_choice_behavior is None
+            or not settings.function_choice_behavior.auto_invoke_kernel_functions
+        ):
+            return await self._send_chat_request(chat_history, settings)
+
+        kernel = kwargs.get("kernel")
+        if not isinstance(kernel, Kernel):
+            raise ServiceInvalidExecutionSettingsError("Kernel is required for auto invoking functions.")
+
+        configure_function_choice_behavior(settings, kernel, update_settings_from_function_choice_configuration)
+
+        for request_index in range(settings.function_choice_behavior.maximum_auto_invoke_attempts):
+            completions = await self._send_chat_request(chat_history, settings)
+            chat_history.add_message(message=completions[0])
+            function_calls = [item for item in chat_history.messages[-1].items if isinstance(item, FunctionCallContent)]
+            if (fc_count := len(function_calls)) == 0:
+                return completions
+
+            results = await invoke_function_calls(
+                function_calls=function_calls,
+                chat_history=chat_history,
+                kernel=kernel,
+                arguments=kwargs.get("argument", None),
```

Comment:
Should this be `arguments` instead of `argument`? The kwarg is added as `arguments` outside of this class.

Reply:
Good catch!